cucumber/features/steps/devLoginPupSteps.js

```diff
@@ -0,0 +1,35 @@
+const {
+  Given, When, Then,
+} = require('@cucumber/cucumber');
+const assertTrue = require('assert');
+const assert = require('assert');
+const scope = require('../support/scope');
+
+Given('the home page of tta-smarthub', async () => {
+  if (!scope.browser) {
+    scope.browser = await scope.driver.launch();
+  }
+  scope.context.currentPage = await scope.browser.newPage();
+  await scope.context.currentPage.goto(scope.uri);
+  await scope.context.currentPage.waitForSelector('a[href$="api/login"]');
+});
+
+Then('we should see "Welcome to the TTA Smart Hub!" message', async () => {
+  await scope.context.currentPage.waitForSelector('h1');
+  const result = await scope.context.currentPage.$('h1');
+  const value = await scope.context.currentPage.evaluate((el) => el.textContent, result);
+
+  assert.equal(value, 'Welcome to the TTA Smart Hub!');
+});
+
+When('pressing login', async () => {
+  await scope.context.currentPage.click('a[href$="api/login"]');
+  await scope.context.currentPage.waitForSelector('.hses');
+});
+
+Then('we should see "Head Start Enterprise System" page', async () => {
+  const result = await scope.context.currentPage.$('.hses');
+  const value = await scope.context.currentPage.evaluate((el) => el.textContent, result);
+
+  assertTrue(value.includes('Head Start Enterprise System'));
+});
```

Comment:
One pattern I've seen work well, is to use `""` in step definitions to mean specific things to look for, so this could be generalized to:

```suggestion
Then('I should see {string} on the page', async (searchString) => {
  // set value to be all the text on the page
  const value = ...
  assertTrue(value.includes(searchString));
});
```

More info on that `{}` format: https://cucumber.io/docs/cucumber/cucumber-expressions/

Similarly, you could create a generic "press button" step and use it for the login step. `When('I press {string}', async (buttonLabel) => {`

Reply:
Yes, most of the future tests include this pattern. Thanks!